cli/kubernetes/options.go

```diff
@@ -202,3 +213,21 @@ func (opts InstallationOptions) ForDeployment(deploymentID string) InstallationO
 
 	return result
 }
+
+// Populate will try to give values to the needed options
+// using the given OptionsReader. If none is given, the default is the
+// InteractiveOptionsReader which will ask in the terminal.
+// This method only populates what is possible and leaves the rest empty.
+// TODO: Implement another method to validate that all options have been set.
+func (opts *InstallationOptions) Populate(reader OptionsReader) (*InstallationOptions, error) {
```

Comment:
I tried to make this method to mutate opts itself but failed. I opted for returning a new pointer. This moves the responsibility of updating the old pointer to the caller, which is something one could forget. Maybe I'm doing something wrong, there must be a way to mutate the receiver (?).

Reply:
Interesting. In the various option readers the mutated structure is a pointer argument, instead of pointer receiver. That certainly works. Maybe somewhere in your code the pointer receiver is a copy of the original structure, thus mutating it does not affect the original ? Because, on the other hand in `ui`, the Message mutators (`WithXXXValue`, etc.) all work.

Reply:
Seeing the code, you are creating a new slice by going through the input slice. How did you try to mutate `opts` itself ?
Something like `opts = newOpts` ? Go has a builtin `copy` function: `copy(opts,newOpts)` ... https://yourbasic.org/golang/copy-explained/

Reply:
Yes I tried `opts = &newOpts`. I will try again when we get all other issues fixed (on this or other PR).